### ISSUE TITLE
[Proposal] Standardized Imitation Learning Pipeline for ./isaaclab.sh --new

### ISSUE THREAD

**Issue by ChaosShi** [isaac-sim/IsaacLab#5561]
Proposal
This proposal requests the addition of a standardized Imitation Learning (IL) pipeline template to the ./isaaclab.sh --new environment creation CLI. Currently, when users create a new project, they are not presented with an option to scaffold a project pre-configured for imitation learning workflows.

The new template should include a complete, production-ready imitation learning structure with native support for data format conversion to mainstream IL frameworks, specifically including:

A ManagerBasedRLMimicEnv-compatible environment that implements the required API methods (get_robot_eef_pose, target_eef_pose_to_action, get_object_poses, get_subtask_term_signals, etc.).

Pre-configured data generation configs (DataGenConfig and SubTaskConfig) for synthetic demonstration augmentation via the MimicGen-style pipeline.

Built-in data format conversion tools enabling seamless conversion of Isaac Lab's HDF5 datasets to:

GR00T LeRobot format (compatible with NVIDIA Isaac GR00T N1.6+)

Standard LeRobot v2/v3 format (compatible with Hugging Face LeRobot ecosystem)

robomimic format (compatible with BC, BC-RNN, HBC, etc.)

Complete IL pipeline scripts: record_demos.py, annotate_demos.py (with auto-annotation support), generate_dataset.py, export_to_groot.py, export_to_lerobot.py, and training scripts.

Proper registry integration ensuring all custom components work seamlessly with Isaac Lab's toolchain.

Motivation
This feature request addresses three core pain points encountered when developing imitation learning applications with Isaac Lab 3.0.

Problem 1: Missing Standard Template

Currently, when running ./isaaclab.sh --new, only RL-focused templates are available. Users must manually study the isaaclab_mimic documentation, implement 6+ API methods, configure SubTaskConfig objects—this is extremely unfriendly for new users.

Problem 2: Broken Toolchain for Registered External Projects

When users register their custom environment as an external extension (the recommended practice), the IL utility scripts—record_demos.py, annotate_demos.py, etc.—fail to run properly due to import path issues, causing a critical workflow breakdown.

Problem 3: Lack of Data Format Support for GR00T and LeRobot Ecosystems

This is the core issue being addressed. Currently, Isaac Lab's IL pipeline outputs standard HDF5 format, but:

NVIDIA Isaac GR00T series models (N1.5, N1.6) use an extended LeRobot-based format that requires a modality.json configuration file to describe state/action array slicing, rotation formats, and normalization methods. The standard LeRobot v2 format lacks this file, preventing the use of GR00T advanced features (semantic splitting, modality normalization, multi-task channels, etc.).

LeRobot ecosystem (Hugging Face) uses Parquet + MP4 based v2/v3 formats, which are completely different from Isaac Lab's HDF5 format.

robomimic uses a specific HDF5 structure. Although Isaac Lab partially supports this, configuration is complex for external projects.

Users are forced to manually write conversion scripts—a tedious and error-prone process (e.g., improper rotation format handling, video/timestamp misalignment). More critically, if developers wish to fine-tune GR00T pretrained models, they already need a server environment with 48GB+ VRAM—data format issues should not be an additional barrier.

Alternatives
Alternative 1: Manual Conversion

Users could write their own conversion scripts. However, the GR00T format requires modality.json to precisely describe slice indices and semantic types for each field—manual configuration is highly error-prone. LeRobot format also requires temporal alignment between video frames and Parquet data, which is complex to implement.

Alternative 2: Community Tools

Community tools like LeIsaac and rdk_LeRobot_tools exist, but they:

Are not officially maintained, with no guarantee of version compatibility

May fall out of sync with Isaac Lab updates

Increase project dependency complexity

Alternative 3: Abandon GR00T Ecosystem

One could use only robomimic for BC training. However, this abandons the powerful vision-language-action joint models of the NVIDIA GR00T series, which is unacceptable for scenarios requiring language-conditioned policies or multimodal inputs.

Build Info
Isaac Lab Version: 3.0.0 (observed on latest main branch as of Feb–Mar 2026)

Isaac Sim Version: 2025.1.0+

Additional Context
Desired User Experience:

bash
# Step 1: Create IL project with template
./isaaclab.sh --new my_il_project --template imitation_learning

cd my_il_project

# Step 2: Record demos + auto-annotate + generate synthetic data
./isaaclab.sh -p scripts/record_demos.py --task MyPickPlaceTask
./isaaclab.sh -p scripts/annotate_demos.py --task MyPickPlaceTask --auto
./isaaclab.sh -p scripts/generate_dataset.py --task MyPickPlaceTask --num_trials 1000

# Step 3: [NEW] Export to GR00T format (for fine-tuning GR00T N1.5/N1.6)
./isaaclab.sh -p scripts/export_to_groot.py \
    --input dataset.hdf5 \
    --output ./groot_dataset/ \
    --modality-config ./configs/groot_modality.json

# Step 4: [NEW] Export to LeRobot format (for Hugging Face ecosystem)
./isaaclab.sh -p scripts/export_to_lerobot.py \
    --input dataset.hdf5 \
    --output ./lerobot_dataset/ \
    --format v3  # or v2

# Step 5: Train (multiple backends supported)
./isaaclab.sh -p scripts/train.py --algo bc_rnn   # robomimic
./isaaclab.sh -p scripts/train_groot.py           # GR00T fine-tuning
Checklist
I have checked that there is no similar issue in the repo (required)

Acceptance Criteria
Criterion 1: Running ./isaaclab.sh --new --template imitation_learning generates a fully functional IL project that runs without modification.

Criterion 2: All required ManagerBasedRLMimicEnv API methods are implemented with sensible defaults.

Criterion 3: record_demos.py, annotate_demos.py, and generate_dataset.py work in external projects without registry or import errors.

Criterion 4: The export_to_groot.py script produces GR00T format datasets that can be directly used for GR00T N1.6 fine-tuning.

Criterion 5: The export_to_lerobot.py script produces LeRobot v2/v3 format datasets loadable via the lerobot library.

Criterion 6: An example modality.json configuration file template is provided covering common robot arm DOF configurations.

Criterion 7: The generated README includes documentation on data format conversion and GR00T/LeRobot integration.

**Comment by RandomOakForest** [comment]
Thank you for posting this. The team will review it.
